### DISCUSSION TITLE
get referer

### DISCUSSION THREAD

**Discussion by breakingbad45** [pocketbase/pocketbase#3620] [Q&A]
is it possible to get referer (which is menioned in log in admin ui) by using this hook onRecordBeforeCreateRequest.i want if users request referer is a specific domain than approve else reject

**Comment by ganigeorgiev** [comment]
_This seems similar to https://github.com/pocketbase/pocketbase/discussions/3619 (if you are the same user please don't open multiple discussions for the same thing and instead try to clarify your question in the existing one)._

With that said, I didn't provide an answer in the related discussion because I don't understand what actually is the desired outcome and I don't know what to suggest.

In general, if you just want to read the referer header from the JS hook you can do it by calling something like [`e.httpContext.request().header.get("Some-Header")`](https://pocketbase.io/docs/js-routing/#reading-request-headers).
If you want to check it from an API rule you can access the header value via [`@request.headers.referer`](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax).

**But there is no guarantee that this header would be available and it can be spoofed so you shouldn't rely on it for security purposes.**

If you want to setup CORS, you can limit the allowed origins with `--origins=domainA.com,domainB.com` flag when starting the executable.

**Reply by apujayed** [reply]
is it possible if i set create rules that @req.auth.id !='' than someone from outside using axios to create if he have any users id password?if yes than how to prevent this kinda situation?

**Reply by ganigeorgiev** [reply]
Sorry, I don't understand what do you mean by _"outside using axios"_.
The rule `@request.auth.id != ''` checks if the request was sent via an authenticated user (_aka. has valid `Authorization: TOKEN` header_). It doesn't know from where and how the request was submitted.

**Reply by apujayed** [reply]
sorry about that..could you please more clarify how to use this“origins=domainA.com,domainB.com”?

**Reply by ganigeorgiev** [reply]
For specifying the allowed CORS origins you can run something like:
```sh
./pocketbase serve --origins=https://domainA.com,https://domainB.com
```
_Edit: see [Origin MDN definition](https://developer.mozilla.org/en-US/docs/Glossary/Origin); `*` and `?` are also allowed._

**Reply by apujayed** [reply]
dont need to mention http or https before domain names?

**Reply by apujayed** [reply]
[Unit]
Description = pocketbase

[Service]
Type           = simple
User           = root
Group          = root
LimitNOFILE    = 4096
Restart        = always
RestartSec     = 5s
StandardOutput = append:/root/pb/errors.log
StandardError  = append:/root/pb/errors.log
ExecStart      = /root/pb/pocketbase serve yourdomain.com

[Install]
WantedBy = multi-user.target 


how to use origin here?

**Reply by ganigeorgiev** [reply]
Add the flag portion at the end of the `ExecStart` line.

The `--origins` flag sets the value for the [`Access-Control-Allow-Origin` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#origin) and yes, based on the documentation it needs to be a valid [origin](https://developer.mozilla.org/en-US/docs/Glossary/Origin):

> Web content's origin is defined by the scheme (protocol), hostname (domain), and port of the [URL](https://developer.mozilla.org/en-US/docs/Glossary/URL) used to access it. Two objects have the same origin only when the scheme, hostname, and port all match.

**Reply by ganigeorgiev** [reply]
_But again just to prevent some eventual misunderstanding - whatever you are trying to do, CORS is primarily a browsers security measure and **will NOT** prevent other users to send requests via curl or other to your API._
